src/gui/points.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+#include <optional>
+#include <iostream>
+#include <cstring>
+#include "3rdparty/Ring-Buffer-master/ringbuffer.hpp"
+#include <QVector>
+#include <QPointF>
+
+typedef enum {Input,Output, None} io ;
```

Comment:
`io` looks a bit too short for a type name, also type names are usually capitalized, so I'd suggest something like `IOType`. Also, typedef like this is not needed in C++. And we can use enum class here.